src/index.ts

```diff
@@ -44,3 +45,62 @@ export function getServiceData(apiName: string, beaconName: string, chain: strin
     },
   };
 }
+
+interface PrivateKeySender {
+  privateKey: string;
+}
+
+interface MnemonicSender {
+  mnemonic: string;
+  derivationPath?: string;
+}
+
+type SenderAccount = PrivateKeySender | MnemonicSender;
+
+export function isPrivateKeySender(senderAccount: SenderAccount): senderAccount is PrivateKeySender {
+  return Object.prototype.hasOwnProperty.call(senderAccount, 'privateKey');
+}
+
+interface WhitelistBeaconReaderResult {
+  beaconId: string;
+  beaconReaderAddress: string;
+  expirationTimestamp: number;
+  indefiniteWhitelistStatus: boolean;
+}
+
+export async function whitelistBeaconReader(
+  beaconId: string,
+  beaconReaderAddress: string,
+  chain: string,
+  providerUrl: string,
+  senderAccount: SenderAccount
+): Promise<WhitelistBeaconReaderResult> {
+  const goBeaconWhitelister = goSync(() =>
+    require(`@api3dao/utility-contracts/SelfServeRrpBeaconServerWhitelister/deployments/${chain}/SelfServeRrpBeaconServerWhitelister.json`)
+  );
+  if (!isGoSuccess(goBeaconWhitelister)) throw new Error(`No whitelister contract found for chain: ${chain}`);
+  const beaconWhitelister = goBeaconWhitelister.data;
+
+  const provider = new ethers.providers.JsonRpcProvider(providerUrl);
+  const wallet = isPrivateKeySender(senderAccount)
```

Comment:
Do we really need to support private keys? I ask because everywhere in API3 code we only rely on mnemonics.

Reply:
Initially we only supported the mnemonic + default derivation path :D  (in admin cli)
Since then we've added optional derivation apth, but Burak didn't like the idea of supporting private keys.

It is easy to support private keys here and some users might want to do so. So I think it's good to keep em here. WDYT?

Reply:
In the years I've been writing code I have learnt (the hard way) that sometimes less is more 😁